homeassistant/components/snapcast/coordinator.py

```diff
@@ -0,0 +1,75 @@
+"""Data update coordinator for Snapcast server."""
+
+from __future__ import annotations
+
+import logging
+
+from snapcast.control.server import Snapserver
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SnapcastUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Data update coordinator for pushed data from Snapcast server."""
+
+    def __init__(self, hass: HomeAssistant, host: str, port: int) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            logger=_LOGGER,
+            name=f"{host}:{port}",
+            update_interval=None,  # Disable update interval as server pushes
+        )
+
+        self._server = Snapserver(hass.loop, host, port, True)
+        self._available = False
+
+        self._server.set_on_update_callback(self._on_update)
+        self._server.set_new_client_callback(self._on_update)
```

Comment:
I guess this is fine. Adds some unnecessary overhead, but usually there won't be thousands of clients.

Reply:
What;s the performance concern? That adding a new client would cause unnecessary state updates for the others?

Reply:
Yes, running on_update with looping through entities when just a client needs to be added is unnecessary. But I guess the overhead is negligible.